backend/Social/SocialAPI.py

```diff
@@ -41,162 +41,70 @@ def hello():
-                    if os.path.exists('profile_pic.gif') == True:
-                        os.remove('profile_pic.gif')
-                        fd = open('profile_pic.gif', 'wb')
-                        fd.write(imgBinary)
-                        fd.close()
-                    else:
-                        fd = open('profile_pic.gif', 'wb')
-                        fd.write(imgBinary)
-                        fd.close()
-                    return imgKey, imgFileLocation
-                elif isWEBP(imgString) == True: 
-                    imgKey = str(userID) + "/profile_pic.webp"
-                    imgFileLocation = str(os.getcwd()) + "/profile_pic.webp"
-                    if os.path.exists('profile_pic.webp') == True:
-                        os.remove('profile_pic.webp')
-                        fd = open('profile_pic.webp', 'wb')
-                        fd.write(imgBinary)
-                        fd.close()
-                    else:
-                        fd = open('profile_pic.webp', 'wb')
-                        fd.write(imgBinary)
-                        fd.close()
-                    return imgKey, imgFileLocation
-                elif isTIFF(imgString) == True: 
-                    imgKey = str(userID) + "/profile_pic.tiff"
-                    imgFileLocation = str(os.getcwd()) + "/profile_pic.tiff"
-                    if os.path.exists('profile_pic.tiff') == True:
-                        os.remove('profile_pic.tiff')
-                        fd = open('profile_pic.tiff', 'wb')
-                        fd.write(imgBinary)
-                        fd.close()
-                    else:
-                        fd = open('profile_pic.tiff', 'wb')
-                        fd.write(imgBinary)
-                        fd.close()
-                    return imgKey, imgFileLocation
+                img_data = binascii.a2b_base64(img_string)
+                if is_jpeg:
+                    img_format = "jpg"
+                elif is_png:
+                    img_format = "png"
+                elif is_gif:
+                    img_format = "gif"
+                elif is_webp:
+                    img_format = "webp"
+                elif is_tiff:
+                    img_format = "tiff"
                 else:
                     raise TypeError
+            img_filename = "profile_pic." + img_format
+            img_key = str(userID) + "/" + img_filename
+            img_file_location = os.getcwd() + "/" + img_filename
+            if os.path.exists(img_filename):
+                os.remove(img_file_location)
+            with open(img_filename, 'wb') as f:
+                f.write(img_data)
+                f.close()
+            return img_key, img_file_location
 
         if not data.get("profilePictureURI"):
-            data["profilePictureURI"] = ""
+            data["profilePictureURI"] = None
+        imgString = data.get("profilePictureURI")
```

Comment:
Can consider modifying this line to `imgString = db.User.find_one({"userID": userID}, {"_id": 0, "passwordHash": 0}).get("profilePictureURI")` and make PUT requests accordingly for migrating existing profile pictures in the form `profilePictureURI`s in the database to S3 purposes.